private-templates-service/client/src/components/Dashboard/PreviewModal/TemplateInfo/TemplateInfo.tsx

```diff
@@ -190,11 +214,35 @@ class TemplateInfo extends React.Component<Props, State> {
             <VersionCard template={this.props.template} templateVersion={this.state.version} />
           </RowWrapper>
         </MainContentWrapper>
-        {this.state.isPublishOpen && getTemplateState(this.props.template, this.state.version) === PostedTemplate.StateEnum.Draft && <PublishModal toggleModal={this.toggleModal} template={this.props.template} templateVersion={this.state.version} />}
-        {this.state.isPublishOpen && getTemplateState(this.props.template, this.state.version) === PostedTemplate.StateEnum.Live && <UnpublishModal toggleModal={this.toggleModal} template={this.props.template} templateVersion={this.state.version} />}
+        {this.props.modalState === ModalState.Publish && <PublishModal template={this.props.template} templateVersion={this.state.version} />}
+        {this.props.modalState === ModalState.Unpublish && <UnpublishModal template={this.props.template} templateVersion={this.state.version} />}
+        {this.props.modalState === ModalState.Share && <ShareModal template={this.props.template} templateVersion={this.state.version} />}
       </OuterWrapper>
     );
   }
 }
 
-export default TemplateInfo;
+function onActionButtonClick(props: Props, state: State, val: any) {
+
+  let templateState = getTemplateState(props.template, state.version);
+
+  if (val.text === 'Share') {
+    props.openModal(ModalState.Share);
+  }
+  else if (val.text === 'Publish') {
+    if (templateState === PostedTemplate.StateEnum.Draft) {
+      props.openModal(ModalState.Publish);
+    }
+    else if (templateState === PostedTemplate.StateEnum.Live) {
+      props.openModal(ModalState.Unpublish);
+    }
+    else {
+      return;
+    }
+  }
+  else {
+    return;
+  }
```

Comment:
`switch (val.text) {
  case 'Share':
    ...
    break;
  case 'Publish':
    switch (templateState) {
      case ....
  }

}`
If you think switch statements are more readable, please refactor this into switch